aten/src/ATen/native/ForeachUtils.h

```diff
@@ -54,5 +54,48 @@ bool check_fast_route(TensorList tensors, Scalar scalar) {
   return true;
 }
 
-} // namespace
+void verify_list(TensorList tensors1, TensorList tensors2) {
+  TORCH_CHECK(tensors1.size() > 0, "Tensor list must have at least one tensor.");
+  TORCH_CHECK(tensors2.size() > 0, "Tensor list must have at least one tensor.");
+  TORCH_CHECK(tensors1.size() == tensors2.size(), "Tensor lists must have the same number of tensors.");
+
+  auto expected_dtype = tensors1[0].dtype();
+
+  for (int i = 0; i < tensors1.size(); i++) {
```

Comment:
The PR body implies that "All tensors in the list must have the same [device]". Should that be checked here?